app/src/main/kotlin/com/simplemobiletools/gallery/pro/activities/ViewPagerActivity.kt

```diff
@@ -1228,7 +1229,8 @@ class ViewPagerActivity : SimpleActivity(), ViewPager.OnPageChangeListener, View
     }
 
     private fun refreshViewPager(refetchPosition: Boolean = false) {
-        if (config.getFolderSorting(mDirectory) and SORT_BY_RANDOM == 0) {
+        val isNotRandomSorting = config.getFolderSorting(mDirectory) and SORT_BY_RANDOM == 0
+        if (isNotRandomSorting || isExternalIntent()) {
```

Comment:
please make this !isRandomSorting, it looks weird

Reply:
[all right :)](https://github.com/SimpleMobileTools/Simple-Gallery/pull/2543/commits/caf6b7c5535f653ad10b997526758c6226d6a505)